[Bug]: AWS provider crashes on AWS Batch setup

### Terraform Core Version

0.13.5

### AWS Provider Version

4.61.0

### Affected Resource(s)

*  aws_batch_compute_environment

### Expected Behavior

Creation of the AWS Batch compute environment resources

### Actual Behavior

AWS plugin crashed

### Relevant Error/Panic Output Snippet

```shell
Stack trace from the terraform-provider-aws_v4.61.0_x5 plugin:
panic: interface conversion: interface {} is nil, not map[string]interface {}
goroutine 420 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/batch.expandComputeResource({0xe8838a0, 0xc0034eaae0}, 0xc0034eb620)
	github.com/hashicorp/terraform-provider-aws/internal/service/batch/compute_environment.go:688 +0xf59
github.com/hashicorp/terraform-provider-aws/internal/service/batch.resourceComputeEnvironmentCreate({0xe8838a0, 0xc0034eaae0}, 0xc004deea80, {0xd384cc0?, 0xc000124800})
	github.com/hashicorp/terraform-provider-aws/internal/service/batch/compute_environment.go:282 +0x465
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xd384cc0?, 0xc000124800?})
	github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:94 +0x175
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xe8838a0?, {0xe8838a0?, 0xc0037453e0?}, 0xd?, {0xd384cc0?, 0xc000124800?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:702 +0x84
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000dbad20, {0xe8838a0, 0xc0037453e0}, 0xc004df15f0, 0xc004dee900, {0xd384cc0, 0xc000124800})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0018a7b30, {0xe8838a0?, 0xc003745230?}, 0xc003a826e0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc0033f5a10, 0xc0033f5a70, {0xc004c6fd00, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.9.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc002563ea0, {0xe8838a0?, 0xc003744330?}, 0xc0017a6690)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xd0899e0?, 0xc002563ea0}, {0xe8838a0, 0xc003744330}, 0xc0017a6620, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0044d34a0, {0xe892fa0, 0xc0005b21a0}, 0xc0047ce240, 0xc004da2630, 0x150d5e00, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0044d34a0, {0xe892fa0, 0xc0005b21a0}, 0xc0047ce240, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.53.0/server.go:963 +0x28a
Error: The terraform-provider-aws_v4.61.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Terraform Configuration Files

```
resource "aws_launch_template" "XXX" {
  name = "XXX"

  # Update the default version to the latest one whenever possible
  update_default_version = true

  capacity_reservation_specification {
    capacity_reservation_preference = "open"
  }

  ebs_optimized = true

  image_id = "XXX"

  instance_initiated_shutdown_behavior = "terminate"

  monitoring {
    enabled = true
  }

  tag_specifications {
    resource_type = "instance"

    tags = {
      Name = "instance"
    }
  }

  user_data = filebase64("test.sh")
}


resource "aws_batch_compute_environment" "XXX" {
  compute_environment_name = "XXX"

  compute_resources {
    instance_role = aws_iam_instance_profile.ecs_instance_role.arn

    instance_type = [
      
    ]

    max_vcpus = X
    min_vcpus = 0

    security_group_ids = [
      aws_security_group.sample.id,
    ]

    subnets = [data.aws_subnet.a.id, data.aws_subnet.b.id, data.aws_subnet.c.id]

    type = "EC2"

    launch_template {
      launch_template_id = aws_launch_template.batch-lauch-template.id
      version            = aws_launch_template.batch-lauch-template.latest_version
    }
  }

  service_role = aws_iam_role.aws_batch_service_role.arn
  type         = "MANAGED"
  depends_on   = [aws_iam_role_policy_attachment.aws_batch_service_role]
}
```

### Steps to Reproduce

Try to set up AWS Batch compute environment using Terraform including a launch template.

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

No

--- a/.changelog/30537.txt
+++ b/.changelog/30537.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_batch_compute_environment: Fix crash when `compute_resources.launch_template` is empty
+```

--- a/internal/service/batch/compute_environment.go
+++ b/internal/service/batch/compute_environment.go
@@ -660,7 +660,7 @@ func expandComputeResource(ctx context.Context, tfMap map[string]interface{}) *b
 		apiObject.InstanceTypes = flex.ExpandStringSet(v)
 	}
 
-	if v, ok := tfMap["launch_template"].([]interface{}); ok && len(v) > 0 {
+	if v, ok := tfMap["launch_template"].([]interface{}); ok && len(v) > 0 && v[0] != nil {
 		apiObject.LaunchTemplate = expandLaunchTemplateSpecification(v[0].(map[string]interface{}))
 	}
 

